### PR TITLE
Bug fix branch

### PR DESCRIPTION
**Pull request by zalavari**
Various bugs indicated by SonarQube fixed

### CODE REVIEW

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java [hunk 1]
```diff
@@ -14,6 +14,8 @@
 
   private int torpedoCount = 0;
 
+  Random generator = new Random();
```
Comment (doliver97) [inline src/main/java/hu/bme/mit/spaceship/TorpedoStore.java]: OK

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java [hunk 2]
```diff
@@ -30,18 +32,17 @@ public TorpedoStore(int numberOfTorpedos){
 
   public boolean fire(int numberOfTorpedos){
     if(numberOfTorpedos < 1 || numberOfTorpedos > this.torpedoCount){
-      new IllegalArgumentException("numberOfTorpedos");
+      throw new IllegalArgumentException("numberOfTorpedos");
```
Comment (doliver97) [inline src/main/java/hu/bme/mit/spaceship/TorpedoStore.java]: OK

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java [hunk 3]
```diff
@@ -30,18 +32,17 @@ public TorpedoStore(int numberOfTorpedos){
...
     boolean success = false;
 
     // simulate random overheating of the launcher bay which prevents firing
-    Random generator = new Random();
     double r = generator.nextDouble();
 
     if (r >= FAILURE_RATE) {
       // successful firing
-      this.torpedoCount =- numberOfTorpedos;
+      this.torpedoCount -= numberOfTorpedos;
```
Comment (doliver97) [inline src/main/java/hu/bme/mit/spaceship/TorpedoStore.java]: Szep munka!

**Review by doliver97 — CHANGES_REQUESTED** [review]
Javitas jovahagyva

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java (hunk 3 above)
Comment (zalavari) [reply src/main/java/hu/bme/mit/spaceship/TorpedoStore.java]: Oké!

**Comment by doliver97** [comment]
Approved

**Pull request merged**